kernel/byterun/coq_interp.c

```diff
@@ -272,6 +272,16 @@ extern void caml_process_pending_signals(void);
 extern double coq_next_up(double);
 extern double coq_next_down(double);
 
+value coq_subst_instance(value arg, value tosubst)
+{
```

Comment:
IDK I blindly copied the code from https://github.com/coq/coq/pull/18903

Reply:
I didn't double-check that code, but I think that to err on the safe side you should use the CAMLparam and CAMLreturn macros in this function. (See the other instances of caml_callback for uint63 arguments.)

Reply:
Yes, it is safe. I am not sure why the code for uint63 bothers registering its arguments as roots. In fact, it is counter-productive. Let me quote a bit of the OCaml runtime: "For best performance, a callback such as `caml_callback_exn(value closure, value arg)` should not extend the lifetime of the values `closure` and `arg` any farther than necessary, that is, they should not be registered as GC roots when the function call actually happens." So, using `CAMLparam` would defeat the optimization provided by the runtime, since it would keep the arguments alive while they might no longer be needed once the callback as completed.

One could be concerned about the very first call, when `caml_named_value` is actually invoked, but fortunately, it is a non-allocating function by design.

Reply:
All right then.